### PR TITLE
Notifications: Adding parsing of heading elements to web notifications

### LINKED ISSUE #45649: Notifications: Expand supported HTML elements.
Masterbar notifications support a very limited set of HTML tags (see `lib/class.notification-builder.phpL125`), causing block rendering to fail from missing elements. Let's expand the supported tags to include `p`, `div`, headings, `cite`, `blockquote`, and `hr` (and any other tags found lacking from blocks).

Screen Shot 2020-09-14 at 12 58 22 PM

Expanding tag support cannot cause regressions with mobile+web notifications.

See: paYKcK-FI-p2

### PR DESCRIPTION
**Pull request by blackjackkent**
#### Changes proposed in this Pull Request

* This pull request, combined with D50168, updates the notifications app to allow heading elements (h1, h2, etc) to be rendered as allowed tags in web notification previews of posts.

#### Testing instructions

* Apply D50168-code in your sandbox and make sure the API is sandboxed.
* Apply this change to calypso, navigate to your Calypso dashboard and open a notification for a post with various heading levels in its content. Verify that the headings are styled distinctly from surrounding paragraphs (for now, just in bold text) and reflected in the markup.

Before:
image
image

After:
image
image

Fixes #45649

### CODE REVIEW

**Comment by blackjackkent** [comment]
This is probably not a blocker to getting this code out the door but I spent a decent chunk of time today trying to solve the related issue mentioned in 198-gh-dotcom-manage regarding header alignment. Header elements that are aligned right or center have a class attribute attached which should be used to style them, but this class attribute is being lost somewhere in the `Notification_Block` rendering logic in the WPCOM code at `wp-content/lib/class.notification-builder.php`. I have tried adding `"class"` as a property to the headings in the `allowed_tags` array that is instantiated in that class:

```php
'h1'         => array(
        'class'   => true,
),
'h2'         => array(
	'class'   => true,
),
'h3'         => array(
	'class'   => true,
),
'h4'         => array(
	'class'   => true,
),
'h5'         => array(
	'class'   => true,
),
'h6'         => array(
	'class'   => true,
),
```

 but I have a sneaking suspicion, at least based on what the code seems to be doing, that the declaration of attributes there doesn't actually change anything in the rendering process, because the alignment classes get lost regardless. Will follow up on this on Monday unless someone can take a stab at it before then.

**Comment by krymson24** [comment]
@blackjackkent tested this and it works as expected! However, it looks like the `h1` is being aligned in the center without styling on my part. I also compared it with `master` and it looks like it doesn't get centered in `master`. I'm not sure if that's related to the comment you made, since your comment is about _preserving_ header alignments. Perhaps it was intentional to make the first text block of the post aligned in the center? See screenshot!

Screen Shot 2020-09-29 at 1 04 27 AM

**Comment by mmtr** [comment]
I also see all `h1` aligned to the center. Seems to be caused by these styles:

https://github.com/Automattic/wp-calypso/blob/8ebfe5ab183555d3055e35b938b26b5f8d03faaf/apps/notifications/src/panel/boot/stylesheets/main.scss#L515-L520

Screen Shot 2020-09-29 at 12 24 25

> the alignment classes get lost regardless

This is fine to investigate in a follow-up PR. Just allowing the heading tags is a big improvement, even if the alignment set on the editor is not preserved.

**Comment by blackjackkent** [comment]
I'm temporarily pulling this back into "In Progress" - I believe the strategy I used for https://github.com/Automattic/wp-calypso/pull/46019 should also allow me to fix alignment for headers as well.

**Comment by kwight** [comment]
I can see the new styles are available and working, but I have to manually add the classes to the elements – it looks like the classes themselves are not getting through in the notif markup.

Screen Shot 2020-09-30 at 1 11 13 PM

**Comment by krymson24** [comment]
Seeing the same as @kwight 
Screen Shot 2020-10-01 at 12 46 27 AM
Screen Shot 2020-10-01 at 12 48 57 AM

File: apps/notifications/style.scss [hunk 1]
```diff
@@ -16,6 +16,23 @@
 		min-width: 0;
 	}
 
+	.has-text-align-center {
```
Comment (kwight) [inline apps/notifications/style.scss:19]: We'll need `has-text-align-left` for RTL languages.

**Comment by blackjackkent** [comment]
Updated with a commit that had not been pushed up! (Plus another to address @kwight 's comment regarding styling.) This should be testable now; sorry for the confusion.

**Comment by kwight** [comment]
Sweet! The left-center-right are all looking good, it's just the `h1`s forced to center (`.wpnc__main .wpnc__single-view h1`) in the notifications styles that remains to fix: 

Screen Shot 2020-10-01 at 11 20 45 AM

**Comment by kwight** [comment]
👍 

Screen Shot 2020-10-01 at 12 10 05 PM

**Review by kwight — APPROVED** [review]

**Pull request merged**